### ISSUE TITLE
When using an empty list in the top file "Received bad data when setting the match..."

### ISSUE THREAD

**Issue by arizvisa** [saltstack/salt#54882]
### Description of Issue
When using an empty list as a placeholder in the topfile, the following error is displayed:
```
2019-10-03 19:44:24,865 [/usr/lib/python2.7/site-packages/salt/matchers/confirm_top.pyc:22  ][ERROR   ][176] Received bad data when setting the match from the top file
```
This happens because the following definition of `salt.matchers.confirm_top` which assumes that an implicit bool means the data is bad. This should probably be checking the type as an empty list is a valid structure to come from yaml. This is a matter of opinion however.

```
def confirm_top(match, data, nodegroups=None):
    '''
    Takes the data passed to a top file environment and determines if the
    data matches this minion
    '''
    matcher = 'compound'
    if not data:
        log.error('Received bad data when setting the match from the top '
                  'file')
        return False
```

### Setup
Define a topfile that looks like the following:
```
base:
  '*':
   []
```

### Steps to Reproduce Issue
You could run anything that processes the topfile, but since it's related you can try `state.show_top`.

### Versions Report
```
Salt Version:
           Salt: 2019.2.1

Dependency Versions:
           cffi: 1.11.5
       cherrypy: Not Installed
       dateutil: 2.8.0
      docker-py: Not Installed
          gitdb: 2.0.3
      gitpython: 2.1.14
          ioflo: 1.7.6
         Jinja2: 2.10.1
        libgit2: 0.27.8
        libnacl: 1.6.1
       M2Crypto: Not Installed
           Mako: 1.1.0
   msgpack-pure: Not Installed
 msgpack-python: 0.6.1
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
   pycryptodome: 3.9.0
         pygit2: 0.27.4
         Python: 2.7.16 (default, Apr 30 2019, 15:54:43)
   python-gnupg: Not Installed
         PyYAML: 5.1
          PyZMQ: 17.0.0
           RAET: 0.6.8
          smmap: 2.0.3
        timelib: Not Installed
        Tornado: 5.0.2
            ZMQ: 4.3.2

System Versions:
           dist: fedora 30 Thirty
         locale: ANSI_X3.4-1968
        machine: x86_64
        release: 4.14.96-coreos
         system: Linux
        version: Fedora 30 Thirty
```

**Comment by arizvisa** [comment]
Closed by #54883.

**Comment by arizvisa** [comment]
Re-visited this as during the upgrade I found these in my logs:

```
2020-06-23 09:52:30,433 [salt.state       :2071][INFO    ][2016] Executing state module.run for [Restart minion with new configuration]
2020-06-23 09:52:30,464 [salt.state       :326 ][INFO    ][2016] {'system.reboot': True}
2020-06-23 09:52:30,464 [salt.state       :2228][INFO    ][2016] Completed state [Restart minion with new configuration] at time 09:52:30.464976 (duration_in_ms=31.239)
2020-06-23 09:52:30,667 [salt.minion      :2154][INFO    ][2016] Returning information for job: req
2020-06-23 09:54:02,492 [c:\python37\lib\site-packages\salt\matchers\confirm_top.py:23  ][ERROR   ][3572] Received bad data when setting the match from the top file
2020-06-23 09:54:02,492 [c:\python37\lib\site-packages\salt\matchers\confirm_top.py:23  ][ERROR   ][3572] Received bad data when setting the match from the top file
2020-06-23 17:13:09,606 [c:\python37\lib\site-packages\salt\matchers\confirm_top.py:23  ][ERROR   ][3572] Received bad data when setting the match from the top file
```
